src/index.tsx

```diff
@@ -658,16 +678,18 @@ class App extends React.Component<{}, AppState> {
 
                 // if no element is clicked, clear the selection and redraw
                 if (draggingElement === null) {
-                  clearSelection();
-                  drawScene();
+                  this.clearSelection();
+                  drawScene(this.state.elements);
                   return;
                 }
 
                 if (elementType === "selection") {
                   if (isDraggingElements) {
                     isDraggingElements = false;
                   }
-                  elements.pop();
+                  const newElements = this.state.elements;
+                  newElements.pop();
```

Comment:
You need to use `newElements.slice(0, -1)` otherwise it mutates the elements array